common/src/main/java/io/pravega/common/io/FileModificationWatcher.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.io;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.function.Consumer;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class can be used to create objects that monitor modifications to the specified file and perform a
+ * specified action (in the form of a callback) upon modification detection.
+ *
+ * Note that:
+ * - The specified actions may trigger slightly later than when the file is actually modified.
+ * - If there are multiple file modifications in quick succession, only the last one may trigger the specified action.
+ *
+ */
+@Slf4j
+public class FileModificationWatcher extends Thread {
```

Comment:
If you really **must** use up an entire thread just for this watcher, please give it a proper name. That way we can filter out by its name in the logs.


Please consider what happens when you have multiple of these running - make sure you don't clash names.

Reply:
Sure, I've added a thread name in the latest commit.